### ISSUE TITLE
Support for Azure Front Door CDN

### ISSUE THREAD

**Issue by a60915** [Azure/AppConfiguration#967]
Does Azure App Configuration support being exposed by an Azure Front Door? We have a security requirement where we must use private networking and a Front Door if we would like to publicly expose a resource.

We have an external resource that we would like to get its configuration from the App Configuration. We have dedicated IPs that would be white-listed in the Front Door. 

I did not see a way to restrict access when using public networking.

**Comment by zhenlan** [comment]
@a60915 you can find a past discussion on the same topic: https://github.com/Azure/AppConfiguration/issues/567.

I'm sure you know you can [enable the private endpoint on your App Configuration store](https://learn.microsoft.com/en-us/azure/azure-app-configuration/concept-private-endpoint) so you can access it from within your vNet.

For public access, we currently do not support client IP restrictions. However, we are working on a Networking Security Perimeter feature that will include client IP restriction and many other security capabilities.

**Comment by a60915** [comment]
@zhenlan Thank you for your response. I had seen topic #567 and read through it. I'm not sure how he setup Frontdoor, but I'm assuming he was keeping the App Config public and was doing this more for availability.

Yes, we are aware of, and using, private endpoints. With public access disabled and private endpoints enabled, I have not found a way to expose the App Configuration using a Front door. There isn't an option for App Configuration when adding an Origin for the Front door. Is there something I am missing or is this not supported?

**Comment by zhenlan** [comment]
Right, we don’t officially support Azure Front Door yet. However, we are working on a feature that will enable you to add App Configuration as an Origin for the Front Door CDN. If everything goes as planned, a preview will be released in the next couple of months.

**Comment by mikejarosch-aw** [comment]
@zhenlan Any updates on supporting App Configuration as an Origin for the Front Door CDN?

**Comment by zhenlan** [comment]
@mikejarosch-aw we ran into a few challenges (technical reasons), so we have revised our design, but we've made good progress. I hope we will have something to share in the next three months.

**Comment by mikejarosch-aw** [comment]
Hi @zhenlan, checking back in. Still on track for releasing something?

**Comment by zhenlan** [comment]
@mikejarosch-aw thanks for checking. Yes, we are on-track to add Azure Front Door CDN support by next month.

**Comment by zhenlan** [comment]
@mikejarosch-aw thank you for checking. We've been actively working on this, but the release has been slightly delayed due to a change in implementation and a few blocking issues we've encountered. We're making good progress, and at this point, we're optimistically targeting a release next month.

**Comment by zhiyuanliang-ms** [comment]
Hey, @mikejarosch-aw 
Sorry for the late response. We sincerely apologize for the delay in AFD support. Due to some security considerations, we had to make a few iterations and updates to both our service backend and client SDK. We also went through some team transitions recently, which affected our bandwidth for a while. The AFD support has been postponed to November. It is now our team’s highest priority, and we’re working hard to deliver it as soon as possible. Thank you so much for your patience and understanding.
